resolving/npm-resolver/src/pickPackageFromMeta.ts

```diff
@@ -58,6 +62,12 @@ export function pickPackageFromMeta (
   }
 }
 
+function pickVersionWithoutDowngrade (meta: PackageMeta, fetchSpec: string, preferredVersionSelectors: VersionSelectors, otherVersion: string): string {
+  const preferredVersions = prioritizePreferredVersions(meta, fetchSpec, preferredVersionSelectors)[0]
```

Comment:
I am not sure using preferredVersionSelectors is a reliable way to get the current version specifier from package.json

Reply:
Does the version specified in the project `package.json` always have the highest priority?

Reply:
Yes, it will have weight set to 1000. However, it can be a direct dependency of another workspace package.

Reply:
So if workspace package `foo` has outdated stable `prettier` and workspace package `bar` has alpha release `prettier`, `prettier` in `foo` won't be updated?

Reply:
But if it was the case, then how could the function `pickPackageFromMeta` picks multiple versions when it only return a single version?

Reply:
> So if workspace package foo has outdated stable prettier and workspace package bar has alpha release prettier, prettier in foo won't be updated?

Yes, I think so. But it would depend on the order in which their version selectors are added to the array.

Reply:
@zkochan I have added two more tests for the workspace use case in https://github.com/pnpm/pnpm/pull/7466/commits/b2dca2b0cd7d9ca3485b2b413c9576541beb5b5d, let me know if it's insufficient to emulate the scenario we are worrying about. If these tests are sufficient, then I believe this is a non-issue.